multimodal_exp_args.py

```diff
@@ -200,15 +204,16 @@ def _setup_devices(self) -> Tuple["torch.device", int]:
             # GPUs available in the environment, so `CUDA_VISIBLE_DEVICES=1,2` with `cuda:0`
             # will use the first GPU in that env, i.e. GPU#1
             device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
-            n_gpu = torch.cuda.device_count()
+            self._n_gpu = torch.cuda.device_count()
```

Comment:
out of scope for this PR: 

original Trainer class should handle all devices for us, it's strange that we are doing this, maybe a task for the future: remove this code and utilize Trainer https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py